src/main/java/io/jenkins/plugins/artifact_manager_jclouds/JCloudsVirtualFile.java

```diff
@@ -363,13 +370,19 @@ private void advanceList(ListContainerOptions options) {
         Deque<CacheFrame> stack = cacheFrames();
         Map<String, CachedMetadata> saved = new HashMap<>();
         int prefixLength = key.length() + /* / */1;
-        listStorageMetadata(true).forEach(sm -> {
-            Long length = sm.getSize();
-            if (length != null) {
-                Date lastModified = sm.getLastModified();
-                saved.put(sm.getName().substring(prefixLength), new CachedMetadata(length, lastModified != null ? lastModified.getTime() : 0));
+        try {
+            Iterator<StorageMetadata> it = listStorageMetadata(true);
```

Comment:
why Iterator instead of Stream? curious

Reply:
I wanted direct handling over exceptions, which is awkward using the `Stream` API.